packages/experience-builder-sdk/src/hooks/useFetchExperience.ts

```diff
@@ -1,41 +1,21 @@
-import type { ContentfulClientApi, Entry } from 'contentful';
-import { useCallback, useEffect, useState } from 'react';
-import { EntityStore } from '../core/preview/EntityStore';
-import { fetchReferencedEntities, fetchExperienceEntry } from '../core/fetchers';
-import type {
-  Experience,
-  ExperienceEntry,
-  ExternalSDKMode,
-} from '@contentful/experience-builder-core/types';
-import { createExperience } from '../utils/createExperience';
-
-const errorMessagesWhileFetching = {
-  experience: 'Failed to fetch experience',
-  experienceReferences: 'Failed to fetch entities, referenced in experience',
-};
-
-const handleError = (generalMessage: string, error: unknown) => {
-  const message = error instanceof Error ? error.message : `Unknown error: ${error}`;
-  console.error(`${generalMessage} with error: ${message}`);
-};
-
-type useClientSideExperienceFetchersProps = {
+import type { ContentfulClientApi } from 'contentful';
+import { useCallback, useState } from 'react';
+import {
+  EntityStore,
+  fetchBySlug as fetchBySlugCore,
+  fetchById as fetchByIdCore,
+} from '@contentful/experience-builder-core';
+import { Experience, ExternalSDKMode } from '@contentful/experience-builder-core/types';
+
+type useClientsideExperienceFetchersProps = {
   mode: ExternalSDKMode;
   client: ContentfulClientApi<undefined>;
-  slug?: string;
-  id?: string;
-  experienceTypeId: string;
-  localeCode: string;
 };
 
-export const useFetchExperience = ({
-  mode,
-  client,
-  slug,
-  id,
-  experienceTypeId,
-  localeCode,
-}: useClientSideExperienceFetchersProps) => {
+/**
+ * @deprecated please use `useFetchBySlug` or `useFetchById` hooks instead
```

Comment:
I guess we can remove this one as well as part of the major release?

Reply:
This will be the first release where it is deprecated. I'd usually wait at least one release to let users move over before it is removed, but since this is alpha its not as big of a deal. Might be good to chat about it as a whole.